File: 2515. Shortest Distance to Target String in a Circular Array/Solution.py
class Solution:
    def closestTarget(self, words: list[str], target: str, startIndex: int) -> int:
        n = len(words)
        for k in range(n // 2 + 1):
            if words[(startIndex - k) % n] == target or words[(startIndex + k) % n] == target:
                return k
        return -1

File: 2515. Shortest Distance to Target String in a Circular Array/Solution.cpp
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int closestTarget(vector<string>& words, string target, int startIndex)
    {
        int n = ssize(words);
        for (int i = 0; i <= n / 2; ++i)
            if (words[(startIndex - i + n) % n] == target || words[(startIndex + i) % n] == target)
                return i;
        return -1;
    }
};
